### PR TITLE
fix(ffe-datepicker-react): fixed issues with focus, tab navigation, k…

### PR DESCRIPTION
**Pull request by ksawerysb1**
…eyboard input

Added new state variables to handle race conditions and tabbing focus changes between components.
Modified event handlers and fixed bugs related to focus flow.

### CODE REVIEW

**Comment by ksawerysb1** [comment]
@selbekk added test for focus and blur of calendar component

File: packages/ffe-datepicker-react/src/datepicker/Datepicker.js [hunk 1]
```diff
@@ -250,9 +253,10 @@ export default class Datepicker extends Component {
     closeCalendarSetInputFocus() {
         this.removeGlobalEventListeners();
         this.setState({
-            openOnFocus: false,
+            openOnFocus: true,
             displayDatePicker: false,
-        });
+            closeAfter: true
+        },this.dateInputRef._input.focus());
```
Comment (selbekk) [inline packages/ffe-datepicker-react/src/datepicker/Datepicker.js]: `this.setState`'s second argument is a function. Pass in an arrow function to make sure `react` can call it whenever it has flushed its state changes:

```js
this.setState(
  {
    openOnFocus: true,
    displayDatePicker: false,
    closeAfter: true,
  }, 
  () => this.dateInputRef._input.focus(),
);
```

**Review by selbekk — DISMISSED** [review]
This looks good to me, but I have to admit I'm a bit shaky as to what's happening here. Would be beneficial to get another person's opinion that has have more to do with this component's source code, before merging.

File: packages/ffe-datepicker-react/src/datepicker/Datepicker.js (hunk 1 above)
Comment (ksawerysb1) [reply packages/ffe-datepicker-react/src/datepicker/Datepicker.js]: Thanks, committed the change now.

**Review by selbekk — DISMISSED** [review]

**Comment by selbekk** [comment]
@henrikhermansen @antidecaf want to have a look?

**Review by selbekk — APPROVED** [review]

**Pull request merged**